machine-learning/02-Data Preparation and Feature Engineering.ipynb

```diff
@@ -1,45 +1,405 @@
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "// Create MLContext\n",
+        "MLContext mlContext = new MLContext();\n",
+        "\n",
+        "//Load Data\n",
+        "IDataView data = mlContext.Data.LoadFromEnumerable<ModelInput>(inMemoryCollection);\n",
+        "\n",
+        "data.Preview(1);"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### What's the difference between a DataFrame and IDataView?\n",
+        "\n",
+        "You may have heard of the [DataFrame](https://docs.microsoft.com/en-us/dotnet/api/microsoft.data.analysis.dataframe?view=ml-dotnet-preview) type. It is another tool to load, view and manipulate data that is common to Notebooks. It implements an IDataView, so it can easily be passed to ML.NET.\n",
+        "\n",
+        "DataFrame and IDataView are very similar in the sense that they both are ways of representing data in a tabular format and applying transformations for it. Some key differences:\n",
+        "\n",
+        "- DataFrame only supports loading delimited files.\n",
+        "- DataFrame runs on memory so you're limited to the amount of memory on your PC.\n",
+        "\n",
+        "The DataFrame is recommended when performing tasks like exploratory data anlysis on a sample of your data. Look at the reference notebook REF - Data Processing for an example of using Data Frames to manipulate a data file for training.\n",
+        "\n",
+        "IDataView is recommended for training on larger datasets, and what we will use here for training examples. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Data Transformations\n",
+        "\n",
+        "ML.NET supports a variety of data transformations that will convert data into the required format and help you make corrections to your data. Some common operations are manipulating columns, normalizing values, replacing missing values, converting values, and more. \n",
+        "\n",
+        "Look for more information on [data transformations available](https://docs.microsoft.com/en-us/dotnet/machine-learning/resources/transforms). \n",
+        "\n",
+        "Below are a few common transformations. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### One Hot Encoding \n",
+        "\n",
+        "One hot encoding is an important step for data containing strings. ML algorithms require data to be numerical, it doesn't know how to process a string. The columns of vendor_id and payment_type are categorical, vendor can be \"CMD\" or \"VST\" and payment can be \"CReDit\" or \"CaSH\". One hot encoding takes the string values passed in and converts them into numerical data."
```

Comment:
I would suggest slightly tweaking to emphasize categorical data.

```suggestion
        "One hot encoding is an important transformation for data containing categories. ML algorithms require data to be numerical, it doesn't know how to process strings representing categories. The columns of vendor_id and payment_type are categorical, vendor can be \"CMD\" or \"VST\" and payment can be \"CReDit\" or \"CaSH\". One hot encoding takes the string values passed in and converts them into numerical data."
```

I'd also add when / why use OneHot vs other techniques. This article does a good job of explaining that (https://www.educative.io/blog/one-hot-encoding). In short.

> Machine learning algorithms treat...a higher number as better or more important than a lower number. This can lead to issues with predictions and poor performance. One hot encoding is useful for data that has no relationship to each other.